### ISSUE TITLE
0.54.3 version does not take into account node roles when checking gateway.expected_nodes

### ISSUE THREAD

**Issue by reversefold** [crate/crate-admin#179]
After an upgrade to 0.54.3 on a cluster with 30 data nodes, 3 master nodes, and 1 client node with `gateway.expected_nodes` set to 33 I'm seeing an error in the crate-admin:
```
The value of the cluster setting 'gateway.expected_nodes' must be equal to the maximum/expected number of nodes in the cluster. 
```
I'm nearly certain that any node which is client-only is not taken into account when `gateway.expected_nodes` is used by ElasticSearch.

**Comment by kovrus** [comment]
@reversefold @joemoe The bug fix is going to be released 0.54.4 hotfix.

**Comment by joemoe** [comment]
@reversefold can you confirm the fix in 0.54.4?

**Comment by reversefold** [comment]
Looks ok in 0.54.4.
